### PR TITLE
Adding snowflake features from IPtProxy

### PR DESCRIPTION
**Pull request by n8fr8**
(also some previous commits from updating the CHANGELOG !)

### CODE REVIEW

File: orbotservice/src/main/java/org/torproject/android/service/OrbotService.java [hunk 1]
```diff
@@ -1414,7 +1429,10 @@ private StringBuffer processSettingsImpl(StringBuffer extraLines) throws IOExcep
             if (meekBridges)
                 extraLines.append("ClientTransportPlugin meek_lite socks5 127.0.0.1:" + IPtProxy.MeekSocksPort).append('\n');
 
-            if (bridgeList != null && bridgeList.length() > 5) //longer then 1 = some real values here
+            if (snowflakeBridges)
+                extraLines.append("ClientTransportPlugin snowflake socks5 127.0.0.1:" + IPtProxy.SnowflakeSocksPort).append('\n');
+
+            if (bridgeList != null && bridgeList.length() > 10) //longer then 1 = some real values here
```
Comment (bitmold) [inline orbotservice/src/main/java/org/torproject/android/service/OrbotService.java:1435]: This comment should be updating. I suppose this works but i wish we had a better way of checking for this

File: orbotservice/src/main/java/org/torproject/android/service/OrbotService.java [hunk 2]
```diff
@@ -326,12 +329,23 @@ private void stopTorAsync() {
 
     }
 
-    private static boolean useIPtProxy ()
+    private static boolean useIPtObfs4Proxy ()
```
Comment (bitmold) [inline orbotservice/src/main/java/org/torproject/android/service/OrbotService.java:332]: This should be renamed since it includes meek as well

**Review by bitmold — COMMENTED** [review]
This looks good! Unfortunately i won't have access to my computer until later this weekend so I haven't tested it yet...

File: orbotservice/src/main/java/org/torproject/android/service/OrbotService.java (hunk 1 above)
Comment (n8fr8) [reply orbotservice/src/main/java/org/torproject/android/service/OrbotService.java:1435]: Yeah, it is really bad. I will take a shot now at improving the whole approach.

File: orbotservice/src/main/java/org/torproject/android/service/OrbotService.java (hunk 2 above)
Comment (n8fr8) [reply orbotservice/src/main/java/org/torproject/android/service/OrbotService.java:332]: Agreed

**Pull request merged**